lib/maths/analytics/CDataFrameCategoryEncoder.cc

```diff
@@ -23,16 +23,24 @@
 #include <maths/common/CBasicStatistics.h>
 #include <maths/common/CChecksum.h>
 
+#include <boost/json.hpp>
 #include <boost/unordered_set.hpp>
 
 #include <algorithm>
 #include <list>
 #include <numeric>
 
+namespace json = boost::json;
+
 namespace ml {
 namespace maths {
 namespace analytics {
 namespace {
+template<typename T>
+bool getEncodingAttribute(const json::object& obj, const std::string& tag, T& value) {
+    std::string tmp{obj.at(tag).as_string().c_str()};
```

Comment:
Why is this going via `c_str()`? Is it because we expect the string representation of `obj` to contain an embedded zero byte and we want to truncate at that point? Whatever the reason is, a comment to explain what's going on would be good.

Reply:
Replaced with the recommended method of converting `json::string` to `std::string`